connector.go

```diff
@@ -22,20 +22,42 @@ type Connector struct {
 	Config   *viper.Viper
 	Health   healthcheck.Handler
 
-	env             kafkautils.Env
-	MsgType         kafkautils.MsgType
-	kafkaUrl        string
-	producerStarted bool
-	consumerStarted bool
+	env      kafkautils.Env
+	MsgType  kafkautils.MsgType
+	kafkaUrl string
 	kafkautils.ProducerInterface
 	*kafkautils.Consumer
 
 	ChainClients     *chain.Clients
 	ProtoRegistryCli *protoregistry.Client
 }
 
-// NewConnector returns a base connector implementation that other connectors can embed to add on to.
-func NewConnector() *Connector {
+func NewProducerConnector() (*Connector, error) {
```

Comment:
it's gonna break other connectors depending on it

Reply:
I will update other connectors right after. I made it so to enforce using other functions after public beta

Reply:
What's the advantage to splitting up new connector into two? What's wrong with the previous automated method

Reply:
There were two issues:
- Must call `ProduceMessage` to start the producer. So it is not possible to order such as Begin - Produce - Commit
- `ProduceMessage` calls `c.producerStarted` for every single write to Kafka. Even if it is a very quick check, it adds up considering we proccess millions of events.

```	
func (c *Connector) ProduceMessage(namespace, subject string, msg proto.Message) error {
 	if !c.producerStarted {
 		err := c.startProducer()
 		if err != nil {
 			return err
 		}
 		c.producerStarted = true
 	}
```